### DISCUSSION TITLE
httpd started root

### DISCUSSION THREAD

**Discussion by hoffdog** [Homebrew/discussions#4575] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.22
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9d01e1deafca35f30b709b5f6a031faaff9642ec
Last commit: 4 days ago
Core tap JSON: 16 Jun 04:26 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit arm_blizzard_avalanche
Clang: 14.0.3 build 1403
Git: 2.41.0 => /opt/homebrew/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4-arm64
CLT: 14.3.1.0.1.1683849156
Xcode: 14.3.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Running ` brew services list `shows the httpd running as root. The User should be my username, not root. Please correct me if I am wrong. I believe this is causing the Forbidden error on a localhost LAMP server. So that you know, I have an additional Intel Macbook Pro running the same setup (LAMP). It shows httpd running as my local username. Any suggestions on how to resolve this are welcome. 

**Things I have tried:**
- brew uninstall httpd
- rm -rf /opt/homebrew/etc/httpd
- brew install httpd
- restarted
- which apachectl
- sudo chmod -R 755 /Users/myusernaem/Sites
- sudo launchctl unload -w /System/Library/LaunchDaemons/org.apache.httpd.plist 2>/dev/null 
- sudo launchctl load -w /System/Library/LaunchDaemons/org.apache.httpd.plist 2>/dev/null 
-  http.conf , changed User to my username, and Group to Admin. Previous was _www

`httpd started root     ~/Library/LaunchAgents/homebrew.mxcl.httpd.plist`

**Comment by hoffdog** [comment]
Happy to provide any additional information.

**Comment by SMillerDev** [comment]
> Running brew services listshows the httpd running as root. The User should be my username, not root.

That depends on how you started it. If you ran `sudo brew services start https` this is correct.

**Reply by hoffdog** [reply]
Make sense. I feel a bit ignorant. Is there a difference between these two commands:

```
brew services start httpd
apachectl start
```

**Reply by SMillerDev** [reply]
`brew services start httpd` runs the command that is defined as the Homebrew service for HTTPD, which on your system is: `/opt/homebrew/opt/httpd/bin/httpd -D FOREGROUND`

`apachectl start` does something else, not sure what since I don't use httpd

**Comment by hoffdog** [comment]
Something else seems to be off. Running ` brew info httpd` shows the DocumentRootas the default path but in the httpd.conf file, it has been updated. 
```
Apache HTTP server
https://httpd.apache.org/
/opt/homebrew/Cellar/httpd/2.4.57 (1,663 files, 32MB) *
  Poured from bottle using the formulae.brew.sh API on 2023-06-16 at 06:58:38
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/httpd.rb
License: Apache-2.0
==> Dependencies
Required: apr ✔, apr-util ✔, brotli ✔, libnghttp2 ✔, openssl@1.1 ✔, pcre2 ✔
==> Caveats
DocumentRoot is /opt/homebrew/var/www.

The default ports have been set in /opt/homebrew/etc/httpd/httpd.conf to 8080 and in
/opt/homebrew/etc/httpd/extra/httpd-ssl.conf to 8443 so that httpd can run without sudo.

To restart httpd after an upgrade:
  brew services restart httpd
Or, if you don't want/need a background service you can just run:
  /opt/homebrew/opt/httpd/bin/httpd -D FOREGROUND
==> Analytics
install: 3,476 (30 days), 12,932 (90 days), 12,932 (365 days)
install-on-request: 3,261 (30 days), 12,162 (90 days), 12,162 (365 days)
build-error: 0 (30 days)
```

**Reply by SMillerDev** [reply]
Yes, caveats are just static text. This is the default setting.
